docs(blog): wow analysis blog post

IndexSeek opened the pull request:
## Description of changes

Adding a new (hopefully fun) blog post analyzing World of Warcraft data to determine who was playing the most, who leveled the fastest, and what activities they did during leveling. 

Original draft on Medium, but hoping we can do this here!

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,174 @@
+---
+title: "Analsis of World of Warcraft Data"
```
```suggestion
title: "Analysis of World of Warcraft Data"
```

IndexSeek replied:
This is addressed in the new commit, thank you! Can't believe I missed that one.

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,174 @@
+  - duckdb
+---
+## Introduction
+This dataset has been around for a while, but it’s newer to me. I grew up playing video games, and I always had a great time jumping into Azeroth and whatever world was introduced in later expansions with my friends.
```
Can you wrap lines? We've got a docs style guide here: https://ibis-project.org/contribute/03_style#documentation-blog-and-other-prose-style

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,174 @@
+
+Ibis ships with an examples module, and this specific data is included. I’ll use DuckDB here, but this is certainly possible with other backends, and I encourage you to experiment. DuckDB is the default Ibis backend, so it’ll be easy to use with this example.
+
+You can execute pip install `ibis-framework[duckdb,examples]` to work with Ibis and the example. You can also throw “jupyter” in to work in IPython or a notebook interface.
```
```suggestion
You can execute `pip install ibis-framework[duckdb,examples]` to work with Ibis and the example data.
```

Mentioning jupyter here like this suggests that `pip install ibis-framework[duckdb,examples,jupyter]` is a thing and since it's not I think that will confuse readers.

IndexSeek replied:
I removed the "jupyter" part to make it clearer.

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,174 @@
+
+### Getting Started
+
+Ibis ships with an examples module, and this specific data is included. I’ll use DuckDB here, but this is certainly possible with other backends, and I encourage you to experiment. DuckDB is the default Ibis backend, so it’ll be easy to use with this example.
```
```suggestion
Ibis ships with an `examples` module, and this specific data is included. I’ll use DuckDB here, but this is certainly possible with other backends, and I encourage you to experiment. DuckDB is the default Ibis backend, so it’ll be easy to use with this example.
```

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,174 @@
+
+### Getting Table Info
+
+I want to learn more about these fields. Are there any nulls we should consider? We can use the info method on our Ibis expression.
```
```suggestion
I want to learn more about these fields. Are there any nulls we should consider? We can use the [`info`](../reference/expression-tables.qmd#ibis.expr.types.relations.Table.info) method on our Ibis expression.
```

This will create a link to the API docs for the `info` method.

cpcloud commented on docs/posts/wow-analysis/index.qmd:
````diff
@@ -0,0 +1,174 @@
+wowah_data.info()
+```
+
+We can also use `value_counts` on specific columns if we want to learn more.
````
```suggestion
We can also use [`value_counts`](../reference/expression-generic.qmd#ibis.expr.types.generic.Column.value_counts) on specific columns if we want to learn more.
```

cpcloud commented on docs/posts/wow-analysis/index.qmd:
````diff
@@ -0,0 +1,174 @@
+```{python}
+wowah_data.mutate(
+    row_num=ibis.row_number().over(
+        ibis.window(group_by=wowah_data.columns, order_by=wowah_data.timestamp)
````
```suggestion
        ibis.window(group_by=wowah_data.columns, order_by="timestamp")
```

For brevity, but totally up to you.

cpcloud replied:
Another way to do this comparison would be to compare

```python
wowah_data.nunique()
```

with

```python
wowah_data.count()
```

IndexSeek replied:
Good catch; I don't refer to the full expr variable name anywhere except to get the columns. I will replace with the `_.timestamp` syntax, so it matches the rest of the code snippets.

cpcloud commented on docs/posts/wow-analysis/index.qmd:
````diff
@@ -0,0 +1,174 @@
+```{python}
+leveler_zones = (
+    level_calc.join(wowah_data, "char", how="inner")
+    .filter((_.timestamp >= _.ts_70) & (_.timestamp <= _.ts_80))
````
```suggestion
    .filter(_.timestamp.between(_.ts_70, _.ts_80 >= _.ts_70))
```

If you want, up to you.

IndexSeek replied:
Thank you for this! I prefer the `between`.

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,174 @@
+    leveler_zones.join(zones, "zone")
+    .group_by([_.char, _.zone_type])
+    .agg(zone_type_count=_.zone.count())
+    .pivot_wider(names_from="zone_type", values_from="zone_type_count")
```
Nice use of `pivot_wider`!

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,174 @@
+
+Doing complicated analysis tasks with Ibis is pretty straightforward. I find that Ibis makes data preparation much easier, especially when trying to apply bulk operations. Breaking things down into steps with these deferred engines makes things much more manageable.
+
+What other questions could we ask of this data?
```
Usually we have a lightweight CTA here (join Zulip, report bug, request features etc), e.g., https://ibis-project.org/posts/duckdb-for-rag/#next-steps.

Up to you! Maybe something like `Get in touch on GitHub, we'd love to see more analyses of this data set`?

IndexSeek replied:
That is a much better CTA! I will change it to this.

```
## Next Steps

Get in touch with us on [GitHub](https://github.com/ibis-project) or 
[Zulip](https://ibis-project.zulipchat.com/), we'd love to see more analyses of this 
data set. 
```

IndexSeek commented:
I just pushed with word wrapping, shortened things up a little bit, rephrased some things, and fixed all curly apostrophes to be straight. 

I also added a thumbnail that I put together for the original Medium article.

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,212 @@
+### Getting Table Info
+
+Let's learn more about these fields. Are there any nulls we should consider? We can
+use the [`info`](../reference/expression-tables.qmd#ibis.expr.types.relations.Table.info)
```
```suggestion
use the [`info`](../../reference/expression-tables.qmd#ibis.expr.types.relations.Table.info)
```

I think these need one more level up!

cpcloud commented on docs/posts/wow-analysis/index.qmd:
````diff
@@ -0,0 +1,212 @@
+wowah_data.info()
+```
+
+We can also use [`value_counts`](../reference/expression-generic.qmd#ibis.expr.types.generic.Column.value_counts)
````
```suggestion
We can also use [`value_counts`](../../reference/expression-generic.qmd#ibis.expr.types.generic.Column.value_counts)
```

cpcloud commented on docs/posts/wow-analysis/index.qmd:
```diff
@@ -0,0 +1,212 @@
+---
+title: "Analsis of World of Warcraft Data"
```
```suggestion
title: "Analysis of World of Warcraft Data"
```

I think this may have gotten clobbered in a subsequent push!

IndexSeek replied:
Thank you! Just fixed, hopefully for the last time on this specific typo haha.
